From b45388559f78aa5ee393340b64f701b9082aaf31 Mon Sep 17 00:00:00 2001
From: Wincent Colaiuta <win@wincent.com>
Date: Fri, 27 Feb 2009 17:16:40 +0100
Subject: [PATCH] Testcase for trailing slash bug for "as" resources

This testcase shows the regression that appeared in Rails 2.3.0
wherein route recognition for resources declared with ":as" was
broken in the presence of a trailing slash.

For example, given a route like:

  map.resources :articles, :as => :wiki

Rails 2.3.0 will raise a RoutingError for the path "/wiki/",
but not for "/wiki". Both variants worked prior to 2.3.0.

See Lighthouse ticket #2039 for more information.

Signed-off-by: Wincent Colaiuta <win@wincent.com>
---
 actionpack/test/controller/routing_test.rb |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/actionpack/test/controller/routing_test.rb b/actionpack/test/controller/routing_test.rb
index 13ba0c3..03d9ab8 100644
--- a/actionpack/test/controller/routing_test.rb
+++ b/actionpack/test/controller/routing_test.rb
@@ -2272,6 +2272,16 @@ class RouteSetTest < Test::Unit::TestCase
     assert_equal("index", request.path_parameters[:action])
   end
 
+  # http://rails.lighthouseapp.com/projects/8994/tickets/2039
+  def test_trailing_slash_with_as_resource
+    set.draw do |map|
+      map.resources :milestones, :as => :roadmap
+    end
+
+    request.path = "/roadmap/"
+    assert_nothing_raised { set.recognize(request) }
+  end
+
   def test_setting_root_in_namespace_using_symbol
     assert_nothing_raised do
       set.draw do |map|
-- 
1.6.1.3

